### PR TITLE
Payments 2.0: full implementation

### PR DESCRIPTION
**Pull request by iTaysonLab**
**features**
- invoice message rendering (both photo/non-photo)
- supported tokenizers: stripe, smartglocal and custom (tested)
- ~99-100% of UI implemented and tested
- support for recurrent payments

**warnings**
- real-life payments are **not tested**, only test ones (via shopbot on prod DC and premium on test DC)
- I tried to make it as stable as it should be, but sometimes bugs can occur (I'm a human after all) - so I'm looking forward to bugreports

**credits**
- @dotvhs for initial mockups

### CODE REVIEW

**Comment by AZIMUT98155** [comment]
> **Особенности**
> 
> * рендеринг сообщения о счете (как фото, так и не фото)
> * поддерживаемые токенизаторы: stripe, smartglocal и custom (проверено)
> * ~99-100% пользовательского интерфейса реализовано и протестировано
> * поддержка рекуррентных платежей
> 
> **предупреждения**
> 
> * реальные платежи **не тестируются** , только тестовые (через шопбот на prod ДЦ и премиум на тестовом ДЦ)
> * Я пытался сделать его настолько стабильным, насколько он должен быть, но иногда могут возникать ошибки (в конце концов, я человек) - так что я с нетерпением жду отчетов об ошибках.
> 
> **кредиты**
> 
> * @dotvhsдля первоначальных макетов

**Comment by vkryl** [comment]
Thank you for your PR. Special build was published at [@tgx_prs](https://t.me/tgx_prs/303) at the time this PR was submitted: https://t.me/tgx_prs/303

There were numerous reports submitted by testers, and, based on commit history, only some of them were addressed. Newer testing builds were never requested, and this PR eventually became abandoned and obsolete, now conflicting with the main branch.

Provided that this feature is security-critical, as it is related to payments, user funds, and work with third-party bots, and the fact that most of your previous PRs required essential advisory and supervision, and yet in long-term they turned out to contain security flaws and bugs that still have to be dealt today, and this PR doesn't seem to be an exception, having to close this PR.

**Comment by iTaysonLab** [comment]
> most of your previous PRs required essential advisory and supervision,
Not "most". You just pretended that you are the "supervisor". Try to get better at "controlling other people", please (for example, start with properly paying people, not trying to scam with "opensource contributions").

Anyway, many thanks for every possibility you all gave to me. That's truly great, and I appreciate that.

Good luck with the development (and I hope you'll finally make a decent feature in a year)!

File: app/src/main/java/org/thunderdog/challegram/component/payments/PaymentFormBottomBarView.java
Comment (TRUPN) [inline app/src/main/java/org/thunderdog/challegram/component/payments/PaymentFormBottomBarView.java:1]: Kenle0303

Comment (elliotthoward) [reply app/src/main/java/org/thunderdog/challegram/component/payments/PaymentFormBottomBarView.java:1]: I need to make money here

**Pull request closed without merging**